File: multiple_linear_regression.R
dataset = read.csv('/Users/sparshgarg/Documents/SantaClara/Machine Learning/Machine Learning_Udemy/Part 2 - Regression/Multiple_Linear_Regression/50_Startups.csv')
dataset$State = factor(x = dataset$State, levels =  c('New York','California','Florida'), labels = c(1,2,3))

library(caTools)
set.seed(123)
split = sample.split(dataset,SplitRatio = 4/5)
training_set = subset(dataset,split==TRUE)
test_set = subset(dataset,split==FALSE)

regressor = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend + State, data = dataset) #to test the whole dataset to check the most
regressor = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend, data = dataset) #to test the whole dataset to check the most
summary(regressor)
regressor = lm(formula = Profit ~ R.D.Spend + Marketing.Spend, data = dataset) #to test the whole dataset to check the most
summary(regressor)
regressor = lm(formula = Profit ~ R.D.Spend, data = dataset) #to test the whole dataset to check the most
summary(regressor)

y_pred = predict(regressor,newdata = test_set)

File: multiple_linear_regression.py
import pandas as pd
import numpy as np

dataset = pd.read_csv('/Users/sparshgarg/Documents/SantaClara/Machine Learning/Machine Learning_Udemy/Part 2 - Regression/Multiple_Linear_Regression/50_Startups.csv')
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 4].values

from sklearn.preprocessing import LabelEncoder,OneHotEncoder
labelencoder = LabelEncoder()
X[:, 3] = labelencoder.fit_transform(X[:, 3])
onehotencoder = OneHotEncoder(categorical_features=[3])
X = onehotencoder.fit_transform(X).toarray()

X = X[:, 1:] #to avoid dummy variable trap

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(X,Y,test_size=0.2,random_state=0)

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train,y_train)

y_pred = regressor.predict(x_test)

#backward elimination
import statsmodels.formula.api as sm
X = np.append(arr=np.ones((50,1)).astype('int'), values= X,axis=1)
X_opt = X[:, [0,1,2,3,4,5]]
regressor_ols = sm.OLS(endog=Y,exog=X_opt).fit()
X_opt = X[:, [0,1,3,4,5]]
regressor_ols = sm.OLS(endog=Y,exog=X_opt).fit()
X_opt = X[:, [0,3,4,5]]
regressor_ols = sm.OLS(endog=Y,exog=X_opt).fit()
X_opt = X[:, [0,3,5]]
regressor_ols = sm.OLS(endog=Y,exog=X_opt).fit()
print(regressor_ols.summary())